Common/src/main/java/com/blamejared/crafttweaker/impl/script/scriptrun/runner/ScriptRunner.java

```diff
@@ -0,0 +1,181 @@
+package com.blamejared.crafttweaker.impl.script.scriptrun.runner;
+
+import com.blamejared.crafttweaker.CraftTweakerCommon;
+import com.blamejared.crafttweaker.api.CraftTweakerAPI;
+import com.blamejared.crafttweaker.api.ICraftTweakerRegistry;
+import com.blamejared.crafttweaker.api.zencode.IScriptLoader;
+import com.blamejared.crafttweaker.api.zencode.scriptrun.IScriptRunInfo;
+import com.blamejared.crafttweaker.api.zencode.scriptrun.IScriptRunModuleConfigurator;
+import com.blamejared.crafttweaker.api.zencode.scriptrun.ScriptRunConfiguration;
+import com.blamejared.crafttweaker.impl.script.scriptrun.natives.CtJavaNativeConverterBuilder;
+import com.google.common.base.Suppliers;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.Util;
+import org.openzen.zencode.java.ScriptingEngine;
+import org.openzen.zencode.java.logger.ScriptingEngineLogger;
+import org.openzen.zencode.java.module.JavaNativeModule;
+import org.openzen.zencode.shared.CompileException;
+import org.openzen.zencode.shared.SourceFile;
+import org.openzen.zenscript.codemodel.FunctionParameter;
+import org.openzen.zenscript.codemodel.SemanticModule;
+import org.openzen.zenscript.lexer.ParseException;
+import org.openzen.zenscript.parser.BracketExpressionParser;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+sealed public abstract class ScriptRunner permits ExecutingScriptRunner, FormattingScriptRunner, SyntaxCheckScriptRunner {
+    
+    @FunctionalInterface
+    private interface ScriptRunCreator {
+        
+        ScriptRunner of(final IScriptRunInfo info, final List<SourceFile> sources, final ScriptingEngineLogger logger);
+        
+    }
+    
+    private static final Supplier<Map<ScriptRunConfiguration.RunKind, ScriptRunCreator>> CREATORS =
+            Suppliers.memoize(() -> Map.of(
+                    ScriptRunConfiguration.RunKind.SYNTAX_CHECK, SyntaxCheckScriptRunner::new,
+                    ScriptRunConfiguration.RunKind.FORMAT, FormattingScriptRunner::new,
+                    ScriptRunConfiguration.RunKind.EXECUTE, ExecutingScriptRunner::new
+            ));
+    
+    private final IScriptRunInfo runInfo;
+    private final List<SourceFile> sources;
+    private final ScriptingEngine scriptingEngine;
+    
+    protected ScriptRunner(final IScriptRunInfo runInfo, final List<SourceFile> sources, final ScriptingEngineLogger logger) {
+        
+        this.runInfo = runInfo;
+        this.sources = List.copyOf(sources);
+        this.scriptingEngine = Util.make(
+                new ScriptingEngine(logger, ScriptRunner.class::getResourceAsStream),
+                it -> it.debug = runInfo.dumpClasses()
+        );
```

Comment:
Why util.make here instead of `this.scriptingEngine.debug = runInfo.dumpClasses()` ?